src/intrinsics/prepack/global.js

```diff
@@ -129,15 +129,49 @@ export default function(realm: Realm): void {
   //     that is not subsequently applied, the function will not be registered
   //     (because prepack won't have a correct value for the FunctionValue itself)
   global.$DefineOwnProperty("__optimize", {
-    value: new NativeFunctionValue(realm, "global.__optimize", "__optimize", 1, (context, [value, config]) => {
+    value: new NativeFunctionValue(realm, "global.__optimize", "__optimize", 1, (context, [value, argModelString]) => {
       // only optimize functions for now
+      if (argModelString !== undefined) {
+        let argModelError;
+        if (argModelString instanceof StringValue) {
+          try {
+            JSON.parse(argModelString.value);
+          } catch (e) {
+            argModelError = new CompilerDiagnostic(
+              "Failed to parse model for arguments",
+              realm.currentLocation,
+              "PP1008",
+              "FatalError"
+            );
+          }
+        } else {
+          argModelError = new CompilerDiagnostic(
+            "String expected as a model",
+            realm.currentLocation,
+            "PP1008",
+            "FatalError"
+          );
+        }
+        if (argModelError !== undefined) {
+          let z = realm.handleError(argModelError);
+          console.log(z);
```

Comment:
This looks like some left-over debug stuff. Also the comment below. Please clean up...